### ISSUE TITLE
Add option to define a Pull button as well

### ISSUE THREAD

**Issue by robert914** [mkloubert/vscode-deploy-reloaded#52]
### Description

Currently, we have the ability to defined a button for the Deploy package files action. It would be helpful for some of my projects if I could also specify a Pull button for the Pull package files action. I suppose it would maybe also be useful if we could specify any number of buttons that correspond to the different actions for Deploy or Pull (current file, package files, all open files).

### Actual behavior

Currently, you can specify a deploy button with:

```json
                "button": {
                    "text": "Deploy to Remote",
                    "tooltip": "Click to deploy files to remote...",
                }
```

### Expected behavior

It would be nice if we could define other action buttons. It would mean your button definition would also need to allow specification of what action to perform. Crude suggestion:

```json
                "buttons": [{
                    "text": "Deploy to Remote",
                    "tooltip": "Click to deploy package files from remote...",
                    "action": "deploy.packagefiles",
                },
                {
                    "text": "Pull from Remote",
                    "tooltip": "Click to pull package files from remote...",
                    "action": "pull.packagefiles",
                }]
```

### Steps to reproduce

Not a bug.

#### Example config

Current style:

```json
                "button": {
                    "text": "Deploy to Remote",
                    "tooltip": "Click to deploy files to remote...",
                }
```

New style example above, but final implementation totally up to you.

#### Logs

Not applicable.

#### Screenshot

Not applicable.

### Your environment

- Operating system: MacOSX 10.12.6
- Visual Studio Code version: 1.21.1
- Extension version: 0.67.0

### Additional comments

Thanks!

**Comment by mkloubert** [comment]
@robert914 

Sorry, when I answer so late.

**I will try to implement such a feature in the upcoming version 0.68.**

The problem is, I have to wait until VSCode 1.22 is relased, because I started to work with new APIs like the upcoming progress notifications:

![preview](https://user-images.githubusercontent.com/4951215/38395547-cff13332-3933-11e8-928f-c590ec212b23.gif)

**Comment by mkloubert** [comment]
@robert914 

Because of the [issue #53](https://github.com/mkloubert/vscode-deploy-reloaded/issues/53) and the new API of VSCode 1.22, which produced a lot of changes, I will try to implement the feature in the next version.

**Comment by mkloubert** [comment]
@robert914 

**I have released a new version 0.69.0**, which can define [global buttons](https://github.com/mkloubert/vscode-deploy-reloaded/wiki/buttons) now:

```json
{
    "deploy.reloaded": {
        "buttons": [
            {
                "text": "Pull PHP files from FTP",
                
                "command": "extension.deploy.reloaded.pullPackgeFiles",
                "arguments": [ "My package", "My FTP server" ]
            }
        ],

        "packages": [
            {
                "name": "My PHP files",
                
                "files": [
                    "**/*.php"
                ]
            }
        ],
        
        "targets": [
            {
                "type": "ftp",
                "name": "My FTP server"
            }
        ]        
    }
}
```

I started a small [wiki page](https://github.com/mkloubert/vscode-deploy-reloaded/wiki/extension_commands), which list intersting commands, provided by the extension, you can use, and explains how to list all available commands, which are currently registrated in VSCode.

**Comment by robert914** [comment]
Thanks a ton! This is awesome! Tons of options to customize it up too!
